app/src/main/java/com/keylesspalace/tusky/MainActivity.java

```diff
@@ -82,14 +85,32 @@
     private static int COMPOSE_RESULT = 1;
 
     private FloatingActionButton composeButton;
-    private String loggedInAccountId;
-    private String loggedInAccountUsername;
     private AccountHeader headerResult;
     private Drawer drawer;
     private ViewPager viewPager;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
+
+        // account switching has to be done before MastodonApi is created in super.onCreate
```

Comment:
I still think that creating multiple Retrofit instances is sub-optimal (caching and scheduling got messed up unless taken care of) and we should set correct URL by subclassing `HttpUrl` and checking for current user URL from there. For notifications we can easily override it with `@Url` parameter or something like this.

Reply:
Yes I know, but thats not a problem introduced with this feature. It works and i would prefer to fix it another time.

Reply:
Yeah, just saying, it shouldn't be in the review